File: graph/BFS/429_nary_tree_level_order_traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if root is None: return []
        res = []
        queue = deque([root])

        while queue:
            temp = []

            for _ in range(len(queue)):
                node = queue.popleft()
                temp.append(node.val)
                queue.extend(node.children)
            res.append(temp)

        return res
        

File: tree/nary_tree/429_n_ary_tree_level_order_traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        if (root == null) return res;

        Queue<Node> q = new LinkedList();
        q.offer(root);
        
        while (!q.isEmpty()) {
            List<Integer> temp = new ArrayList();
            int size = q.size();
            for (int i = 0; i < size; i++){
                Node node = q.poll();
                temp.add(node.val);
                q.addAll(node.children);
            }
            res.add(temp);
        }
        return res;
    }
}